core/common/src/ContourDisplay.ts

```diff
@@ -0,0 +1,296 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Symbology
+ */
+
+import { compareBooleans, compareNumbers, CompressedId64Set, OrderedId64Iterable } from "@itwin/core-bentley";
+import { ColorDef, ColorDefProps } from "./ColorDef";
+import { LinePixels } from "./LinePixels";
+
+/** JSON representation of the style settings used by either a minor or major contour. */
+export interface ContourStyleProps {
+  color?: ColorDefProps;
+  pixelWidth?: number;
+  pattern?: LinePixels;
+}
+
+/** The style settings used by either a minor or major contour.
+   * @see [[Contour.majorStyle]]
+   * @see [[Contour.minorStyle]]
+   */
+export class ContourStyle {
```

Comment:
Add `clone` methods to the various immutable types in this file following the pattern used elsewhere, accepting a `Partial<WhateverProps>`, so people can create modified copies.

Reply:
As part of attempting to resolve [this comment chain](https://github.com/iTwin/itwinjs-core/pull/7205#discussion_r1792386401), I began working on adding this to all the other ContourDisplay structures --- namely ContourGroup, Contour, and ContourStyle.

However, I ran into difficulty with ContourGroup and allowing it to use the Partial paradigm, particularly the way subCategories are exposed as an OrderedId64Iterable but stored internally as a CompressedId64Set.

Do you have any suggestions?

Otherwise, we could just skip this for now because at least ContourDisplay now supports creating itself with partial properties, namely the displayContours flag.

Reply:
I can't tell if you did this or not, or what specific difficulties you encountered. People (e.g., Booster) who want to adjust the current contour settings will need these methods.

Reply:
this was pushed this morning, and I added tests for it.